File: src/main/java/cherkasov/com/streamsource/DummyConnection.java
package cherkasov.com.streamsource;

import java.io.IOException;
import java.io.InputStream;

/**
 * Dummy
 * For test purposes only
 * All methods return false or null
 */

public class DummyConnection implements Connection {
    @Override
    public long getContentLength() {
        return 0;
    }

    @Override
    public InputStream getInputStream() throws IOException {
        return null;
    }

    @Override
    public void disconnect() {
    }

    @Override
    public boolean isConnected() {
        return false;
    }

    @Override
    public boolean connect() {
        return false;
    }
}

File: src/main/java/cherkasov/com/streamsource/DummyConnection.kt
package cherkasov.com.streamsource

import java.io.IOException
import java.io.InputStream

/**
 * Dummy
 * For test purposes only
 * All methods return false or null
 */
class DummyConnection : Connection {
    override val contentLength: Long
        get() = 0

    @get:Throws(IOException::class)
    override val inputStream: InputStream?
        get() = null

    override fun disconnect() {}
    override val isConnected: Boolean
        get() = false

    override fun connect(): Boolean {
        return false
    }
}